analytics/service_test.go

```diff
@@ -17,11 +25,352 @@ func Test_extractIntParam(t *testing.T) {
+
+		url, err := s.CaptureAnalyticsData(r)
+		So(err, ShouldBeNil)
+		So(url, ShouldEqual, "/peoplepopulationandcommunity")
+	})
+
+	Convey("Given a valid redirect URL, and aud has no elements in []string", t, func() {
+		r, err := http.NewRequest("GET", "/redir/{:data}", nil)
+		So(r, ShouldNotBeNil)
+		So(err, ShouldBeNil)
+
+		q := r.URL.Query()
+
+		// Create a new token object, specifying signing method and the claims
+		// you would like it to contain.
+		token := jwt.NewWithClaims(jwt.SigningMethodHS256, jwt.MapClaims{
+			"index": 1,
+			"aud":   []string{},
+			"uri":   "/peoplepopulationandcommunity",
+		})
+
+		// Sign and get the complete encoded token as a string using the secret
+		tokenString, err := token.SignedString(hmacSampleSecret)
+		So(err, ShouldBeNil)
+
+		q.Set(":data", tokenString)
+		r.URL.RawQuery = q.Encode()
+
+		url, err := s.CaptureAnalyticsData(r)
+		So(err, ShouldBeNil)
+		So(url, ShouldEqual, "/peoplepopulationandcommunity")
+	})
+
+	Convey("Given a valid redirect URL, and aud has one element in []string", t, func() {
+		r, err := http.NewRequest("GET", "/redir/{:data}", nil)
+		So(r, ShouldNotBeNil)
+		So(err, ShouldBeNil)
+
+		q := r.URL.Query()
+
+		// Create a new token object, specifying signing method and the claims
+		// you would like it to contain.
+		token := jwt.NewWithClaims(jwt.SigningMethodHS256, jwt.MapClaims{
+			"index": 1,
+			"aud":   []string{"hello2"},
+			"uri":   "/peoplepopulationandcommunity",
+		})
+
+		// Sign and get the complete encoded token as a string using the secret
+		tokenString, err := token.SignedString(hmacSampleSecret)
+		So(err, ShouldBeNil)
+
+		q.Set(":data", tokenString)
+		r.URL.RawQuery = q.Encode()
+
+		url, err := s.CaptureAnalyticsData(r)
+		So(err, ShouldBeNil)
+		So(url, ShouldEqual, "/peoplepopulationandcommunity")
+	})
+
+	Convey("Given a valid redirect URL, and aud has two elements in []string", t, func() {
+		r, err := http.NewRequest("GET", "/redir/{:data}", nil)
+		So(r, ShouldNotBeNil)
+		So(err, ShouldBeNil)
+
+		q := r.URL.Query()
+
+		// Create a new token object, specifying signing method and the claims
+		// you would like it to contain.
+		token := jwt.NewWithClaims(jwt.SigningMethodHS256, jwt.MapClaims{
+			"index": 1,
+			"aud":   []string{"hello3", "hello4"},
+			"uri":   "/peoplepopulationandcommunity",
+		})
+
+		// Sign and get the complete encoded token as a string using the secret
+		tokenString, err := token.SignedString(hmacSampleSecret)
+		So(err, ShouldBeNil)
+
+		q.Set(":data", tokenString)
+		r.URL.RawQuery = q.Encode()
+
+		url, err := s.CaptureAnalyticsData(r)
+		So(err, ShouldBeNil)
+		So(url, ShouldEqual, "/peoplepopulationandcommunity")
+	})
+}
+
+// The following code is borrowed from v3.2.2 of alternative library:
+// github.com/form3tech-oss/jwt-go
+// and tweaked a little to demonstrate issues with original library:
+// github.com/dgrijalva/jwt-go
+// BUT the code won't work with v4 of original library
```

Comment:
This comment should probably be reworded slightly. It implies knowledge of this change (this PR) when talking about "alternative" and "original" libraries. But this comment will be referring to the current code in two, three weeks/months/years time with no knowledge of what was here before. So the comment should refer to the code in place. Eg, 
```suggestion
// The following code is borrowed from v3.2.2 of library:
// github.com/form3tech-oss/jwt-go
// and tweaked a little to demonstrate issues with the previously used library:
// github.com/dgrijalva/jwt-go
// BUT the code won't work with v4 of original library
```